src/irc/ClientPool.ts

```diff
@@ -558,8 +558,10 @@ export class ClientPool {
             for (const [userId, client] of set.userIds.entries()) {
                 try {
                     const banReason = this.ircBridge.matrixBanSyncer?.isUserBanned(userId);
-                    log.warn(`Killing ${userId} client connection due - user is banned (${banReason})`);
-                    await client.kill('User was banned');
+                    if (banReason) {
```

Comment:
I'd prefer some test that also kills on empty string.
While the `matrixBanSyncer` will default empty strings to "No reason given", only the return value `false` states that the user isn't banned.

```suggestion
                    if (typeof banReason === 'string') {
```